homeassistant/components/particle/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""Integration with the Particle Cloud."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_ACCESS_TOKEN, STATE_ON, STATE_UNAVAILABLE
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import make_entity_service_schema
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = "particle"
+
+_LOGGER = logging.getLogger(DOMAIN)
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.Schema({vol.Required(CONF_ACCESS_TOKEN): cv.string})},
+    extra=vol.ALLOW_EXTRA,
+)
+
+ATTR_ARGS = "args"
+ATTR_FUNCTION = "function"
+
+FUNCTION_CALL_SCHEMA = make_entity_service_schema(
+    {
+        vol.Required(ATTR_FUNCTION): cv.string,
+        vol.Optional(ATTR_ARGS): vol.All(
+            cv.ensure_list, vol.Length(min=1), [cv.string]
+        ),
+    }
+)
+
+
+async def async_setup(hass, config):
+    """Initialize our component while Home Assistant loads."""
+    if DOMAIN not in config:
+        return True
+
+    from pyparticleio.ParticleCloud import ParticleCloud
```

Comment:
Please move all imports to the top.

Reply:
So, let's talk about this. As far as earlier reviews were concerned, this PR was ready to go. Is there a written-down process for getting these merged?

The style guide explicitly requests modules only be loaded when required, which is why the `import` statement is here. What about conflicts between these additional comments, and the style?

I'd like clarity on where we go from here before I put more time into this code.

Thanks.

Reply:
That rule has been lifted quite some time ago, the style guideline requiring the import when it was used, has been removed at that time as well.

Please move the imports to the top (as we did quite a big effort to move everything to the top for the hundreds of integrations we have 😉)